### PR TITLE
wrap UnknownHostException to ConfigException

### PR DESCRIPTION
**Pull request by vietnguyen-td**
We have a retry mechanism, when starting a job with an invalid configuration, it should not be retry.
This PR to wrap UnknownHostException to ConfigException when starting `transaction`

### CODE REVIEW

**Review by kietdo360 — APPROVED** [review]
LGTM

File: embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java [hunk 1]
```diff
@@ -218,6 +220,10 @@ public ConfigDiff transaction(ConfigSource config,
             // TODO incremental_columns is not set => get primary key
             schema = setupTask(con, task);
         } catch (SQLException ex) {
+            final Throwable cause = ex.getCause();
+            if (cause instanceof SocketTimeoutException || cause instanceof UnknownHostException) {
```
Comment (dmikurube) [inline embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java:224]: It is understandable `UnknownHostException` to consider as `ConfigException`. But `SocketTimeoutException` does not sound like a `ConfigException` actually.

(Even when we consider a retry based on a type of Exception (sorry, it's our internal use-case, @hito4t), socket timeout is considered to recover by some retries?)

Comment (vietnguyen-td) [reply embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java:224]: When i am testing on local with invalid host, i also get both these exceptions with invalid `host`
- `UnknownHostException` when internet issue,  unable to resolve internet addresses
- `SocketTimeoutException` when server issue, fail to response

imo, those exceptions have chance to recover by some retries, we ignore only when start `transaction`

Comment (dmikurube) [reply embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java:224]: The problem here is NOT that you received those exceptions when testing on your local environments. The problem here is that real tentative network unstability is also (unintentionally or unexpectedly) affected by this.

Comment (vietnguyen-td) [reply embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java]: removed SocketTimeoutException

**Review by dmikurube — APPROVED** [review]
Okay, looks good to me!

@hito4t Unless you have any concern, I think we'll merge this early next week.

**Review by hito4t — APPROVED** [review]

**Review by ylzen — APPROVED** [review]
Reviewed for Data Exfiltrations.

**Comment by dmikurube** [comment]
Resolved conflicts. Merging after confirming the CI is okay.

**Pull request merged**